pinot-core/src/main/java/org/apache/pinot/core/segment/index/readers/text/LuceneTextIndexReader.java

```diff
@@ -0,0 +1,139 @@
+   * the reference in {@link org.apache.pinot.core.segment.index.column.PhysicalColumnIndexContainer}
+   * similar to how it is done for other types of indexes.
+   * @param column column name
+   * @param segmentIndexDir segment index directory
+   */
+  public LuceneTextIndexReader(String column, File segmentIndexDir) {
+    _column = column;
+    try {
+      File indexFile = new File(segmentIndexDir.getPath() + "/" + SegmentVersion.v3.name() + "/" + column + LuceneTextIndexCreator.LUCENE_TEXT_INDEX_FILE_EXTENSION);
+      _indexDirectory = FSDirectory.open(indexFile.toPath());
+      _indexReader = DirectoryReader.open(_indexDirectory);
+      _indexSearcher = new IndexSearcher(_indexReader);
+      // Disable Lucene query result cache. While it helps a lot with performance for
+      // repeated queries, on the downside it cause heap issues.
+      _indexSearcher.setQueryCache(null);
+    } catch (Exception e) {
+      LOGGER.error("Failed to instantiate Lucene text index reader for column {}, exception {}", column, e.getMessage());
+      throw new RuntimeException(e);
+    }
+    StandardAnalyzer analyzer = new StandardAnalyzer();
+    _queryParser = new QueryParser(column, analyzer);
+  }
+
+  @Override
+  public LuceneSearchResult getDocIds(int dictId) {
+    // This should not be called from anywhere. If it happens, there is a bug in the current implementation
+    // and that's why we throw illegal state exception
+    throw new IllegalStateException("Using dictionary ID is not supported on Lucene inverted index");
+  }
+
+  /**
+   * Called during filter operator execution
+   * by {@link org.apache.pinot.core.operator.filter.TextMatchFilterOperator}
+   * @param predicate text search predicate
+   * @return search results
+   */
+  @Override
+  public LuceneSearchResult getDocIds(Predicate predicate) {
+    String searchQuery = ((TextMatchPredicate)predicate).getSearchQuery();
+    MutableRoaringBitmap docIDs = new MutableRoaringBitmap();
+    Collector docIDCollector = new LuceneDocIdCollector(docIDs);
+    try {
+      Query query = _queryParser.parse(searchQuery);
+      _indexSearcher.search(query, docIDCollector);
+      return new LuceneSearchResult(docIDs, new LuceneIndexSearcherReferenceManager(null, _indexSearcher));
+    } catch (Exception e) {
+      LOGGER.error("Failed while searching the text index for column {}, search query {}, exception {}",
+          _column, searchQuery, e.getMessage());
+      throw new RuntimeException(e);
+    }
+  }
+
+  /**
+   * When we destroy the loaded ImmutableSegment, all the indexes
+   * (for each column) are destroyed and as part of that
+   * we release the text index
+   * @throws IOException
+   */
+  @Override
+  public void close() throws IOException {
+    _indexReader.close();
+    _indexDirectory.close();
+  }
+
+  public static class LuceneSearchResult {
+    final MutableRoaringBitmap luceneDocIDs;
+    final LuceneIndexSearcherReferenceManager referenceManager;
```

Comment:
Why do you need to keep reference count for inverted index result? The bitmap is new created right?

Reply:
I have actually removed this class. All the logic is now encapsulated in the reader